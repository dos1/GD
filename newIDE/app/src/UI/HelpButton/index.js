// @flow
import React from 'react';
import FlatButton from 'material-ui/FlatButton';
import HelpOutline from 'material-ui/svg-icons/action/help-outline';
import Window from '../../Utils/Window';
import { getHelpLink } from '../../Utils/HelpLink';

type PropsType = {
  helpPagePath: ?string,
  label?: string,
};

/**
 * The button that can be used in any dialog to open a help page
 */
const HelpButton = (props: PropsType) => {
  if (!props.helpPagePath) return null;

  return (
    <FlatButton
      onClick={() => {
        if (props.helpPagePath) {
          Window.openExternalURL(getHelpLink(props.helpPagePath));
        }
      }}
      target="_blank"
      label={props.label || 'Help'}
      icon={<HelpOutline />}
    />
  );
};

export default HelpButton;
