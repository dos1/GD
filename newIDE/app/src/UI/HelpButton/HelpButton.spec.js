// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import HelpButton from '.';

describe('HelpButton', () => {
  it('renders the button linking to a help page', () => {
    const tree = renderer
      .create(
        <MuiThemeProvider>
          <HelpButton helpPagePath="/objects/tiled_sprite" />
        </MuiThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders nothing if the helpPagePath is empty', () => {
    const tree = renderer
      .create(
        <MuiThemeProvider>
          <HelpButton helpPagePath="" />
        </MuiThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
