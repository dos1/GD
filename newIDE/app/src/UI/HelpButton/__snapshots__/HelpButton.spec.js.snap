// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpButton renders nothing if the helpPagePath is empty 1`] = `null`;

exports[`HelpButton renders the button linking to a help page 1`] = `
<button
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
      "backgroundColor": "rgba(0, 0, 0, 0)",
      "border": 10,
      "borderRadius": 2,
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "Roboto, sans-serif",
      "fontSize": "inherit",
      "fontWeight": "inherit",
      "height": 36,
      "lineHeight": "36px",
      "margin": 0,
      "minWidth": 88,
      "muiPrepared": true,
      "outline": "none",
      "overflow": "hidden",
      "padding": 0,
      "position": "relative",
      "textAlign": "center",
      "textDecoration": "none",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "userSelect": "none",
      "verticalAlign": null,
    }
  }
  tabIndex={0}
  target="_blank"
  type="button"
>
  <div
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
  >
    <svg
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "rgba(0, 0, 0, 0.87)",
          "height": 24,
          "marginLeft": 12,
          "marginRight": 0,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": "middle",
          "width": 24,
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
      />
    </svg>
    <span
      style={
        Object {
          "fontSize": 14,
          "fontWeight": 500,
          "letterSpacing": 0,
          "muiPrepared": true,
          "paddingLeft": 8,
          "paddingRight": 16,
          "position": "relative",
          "textTransform": "uppercase",
          "verticalAlign": "middle",
        }
      }
    >
      Help
    </span>
  </div>
</button>
`;
